process/transactions/scDeploysProcessor.go

```diff
@@ -0,0 +1,232 @@
+package transactions
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/elastic-indexer-go/data"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/hashing/keccak"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+)
+
+const (
+	delegationManagerAddress = "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6"
+)
+
+type scDeploysProc struct {
+	pubKeyConverter      core.PubkeyConverter
+	scDeployReceiverAddr string
+	selfShardID          uint32
+}
+
+type deployerDto struct {
+	receiver string
+	sender   string
+	nonce    uint64
+	data     []byte
+	txHash   string
+}
+
+func newScDeploysProc(pubKeyConverter core.PubkeyConverter, selfShardID uint32) *scDeploysProc {
+	scDeployReceiver := make([]byte, 32)
+
+	return &scDeploysProc{
+		selfShardID:          selfShardID,
+		pubKeyConverter:      pubKeyConverter,
+		scDeployReceiverAddr: pubKeyConverter.Encode(scDeployReceiver),
+	}
+}
+
+func (sc *scDeploysProc) searchSCDeployTransactionsOrSCRS(
+	txs []*data.Transaction,
+	scrs []*data.ScResult,
+) []*data.ScDeployInfo {
+	scDeploys := make([]*data.ScDeployInfo, 0)
+	for _, tx := range txs {
+		dto := &deployerDto{sender: tx.Sender, receiver: tx.Receiver, nonce: tx.Nonce, txHash: tx.Hash, data: tx.Data}
```

Comment:
This line is duplicated with L82. Normally, won't be such an issue for just one line. In this case it is very important because we might add in the future a new field that will be set only in one of this function. Lots of duplicated code from this repo are originated by the fact that you are using plain old pointers for data.Transaction, data.ScResult and so on. Instead, if you would have created an interface on the model of the TransactionHandler found in elrond-go then the code would have looked better.

Reply:
Added a task where will implement your suggestion.  elasticsearch Transaction and ScResult with implement the same interface in order to remove duplicated code. **[EN-9843]**